app/controllers/cloud_subnet_controller.rb

```diff
@@ -235,15 +232,15 @@ def new_form_params
     options[:name] = params[:name] if params[:name]
     options[:ems_id] = params[:ems_id] if params[:ems_id]
     options[:cidr] = params[:cidr] if params[:cidr]
-    # An address is automatically assigned by Openstack when gateway is null
+    # Provider to automatically assign gateway address unless provided
     if params[:gateway]
       options[:gateway_ip] = params[:gateway].blank? ? nil : params[:gateway]
     end
     options[:ip_version] = params[:network_protocol] =~ /4/ ? 4 : 6
-    options[:cloud_tenant] = find_record_with_rbac(CloudTenant, params[:cloud_tenant_id]) if params[:cloud_tenant_id]
+    options[:cloud_tenant] = find_record_with_rbac(CloudTenant, params[:cloud_tenant][:id]) if params[:cloud_tenant][:id]
```

Comment:
Please use `params.fetch_path(:cloud_tenant, :id)` to prevent `undefined method `[]' for nil:NilClass` error.

Reply:
This should be fixed ☝️